micromamba/tests/test_create.py

```diff
@@ -665,14 +663,96 @@ def test_create_empty(tmp_home, tmp_root_prefix, tmp_path, prefix_selector, crea
+def test_create_envs_dirs(tmp_root_prefix: Path, tmp_path: Path, monkeypatch):
     """Create an environment when the first env dir is not writable."""
-    os.environ["CONDA_ENVS_DIRS"] = f"{Path('/noperm')},{tmp_path}"
+
+    noperm_root_dir = Path(tmp_path / "noperm")
+    noperm_envs_dir = noperm_root_dir / "envs"
+
+    monkeypatch.setenv("CONDA_ENVS_DIRS", f"{noperm_envs_dir},{tmp_path}")
     env_name = "myenv"
-    helpers.create("-n", env_name, "--offline", "--no-rc", no_dry_run=True)
+    os.makedirs(noperm_root_dir, exist_ok=True)
+
+    if platform.system() == "Windows":
+        subprocess.run(
+            [
+                "icacls",
+                noperm_root_dir,
+                "/inheritance:r",
+                "/grant:r",
+                "Everyone:(RX)",
+                "/remove",
+                r"BUILTIN\Administrators",
+                "/remove",
+                r"NT AUTHORITY\SYSTEM",
+                "/remove",
+                "OWNER RIGHTS",
+            ],
+            check=True,
+        )
+
+    else:
+        os.chmod(noperm_root_dir, 0o555)
+
+    try:
+        helpers.create("-n", env_name, "--offline", "--no-rc", no_dry_run=True)
+    finally:
+        if platform.system() == "Windows":
+            subprocess.run(
+                ["icacls", noperm_root_dir, "/grant", r"BUILTIN\Administrators:F"], check=True
+            )
+            subprocess.run(["icacls", noperm_root_dir, "/reset"], check=True)
+        else:
+            os.chmod(noperm_root_dir, 0o755)
+
     assert (tmp_path / env_name / "conda-meta" / "history").exists()
 
 
+@pytest.mark.parametrize("envs_dirs_source", ("condarc", "env_var"))
+def test_mkdir_envs_dirs(tmp_path, tmp_home, monkeypatch, envs_dirs_source):
+    """Test that an env dir is created if it does not exist already"""
+
+    envs_dir = tmp_path / "user_provided_envdir" / "envs"
+
+    with open(tmp_home / ".condarc", "w+") as f:
+        if envs_dirs_source == "env_var":
+            monkeypatch.setenv("CONDA_ENVS_DIRS", str(envs_dir))
+        else:
+            f.write(f"envs_dirs: [{str(envs_dir)}]")
+
+    assert not envs_dir.exists()  # directory doesn't exist yet
+
+    helpers.create("-n", "bar", "--rc-file", tmp_home / ".condarc", no_rc=False)
+
+    assert envs_dir.exists()
+
+
+def test_env_dir_idempotence(tmp_path, tmp_home, monkeypatch):
+    """
+    Test that setting envs_dirs to ~/.conda and running twice in a row
+    gives the same results
+    https://github.com/mamba-org/mamba/issues/3836
+    """
+
+    mamba_root_prefix = tmp_path / "envroot"
+    mamba_root_prefix_envs = mamba_root_prefix / "envs"
+    condarc_envs_dirs = tmp_home / ".conda"
+
+    with open(tmp_home / ".condarc", "w+") as f:
+        f.write(f"envs_dirs: [{str(condarc_envs_dirs)}]")
+
+    os.makedirs(mamba_root_prefix, exist_ok=True)
+    monkeypatch.setenv("MAMBA_ROOT_PREFIX", str(mamba_root_prefix))
+    env_name = "foo"
```

Comment:
I believe this can be simplified using `tmp_root_prefix` fixture:
```suggestion
    mamba_root_prefix_envs = tmp_root_prefix / "envs"
    condarc_envs_dirs = tmp_home / ".conda"

    with open(tmp_home / ".condarc", "w+") as f:
        f.write(f"envs_dirs: [{str(condarc_envs_dirs)}]")

    env_name = "foo"
```